### ISSUE TITLE
On application start, flush all meetings and data on Mongo

### ISSUE THREAD

**Issue by antobinary** [bigbluebutton/bigbluebutton#5079]
Currently old meetings/users are kept on restart of `bbb-html5.service`. It must be the case for running meteor manually too

**Comment by OZhurbenko** [comment]
Wasn't the goal to specifically keep everything in Mongo on restart?
As you remember we used to delete everything from Mongo on restart, but then @oswaldoacauan and @Gcampes decided to change that.
Oswaldo spent a lot of time making sure old data gets properly replaced with the new data on restart.

**Comment by antobinary** [comment]
Perhaps I am seeing issues with clearing meetings that have ended. The symptom I see is high count of meetings and offline users on a server with few running meetings. I see the benefit on preserving state on a development environment (while making server side changes for example) but do not see it on production. Please comment.

**Comment by OZhurbenko** [comment]
I, personally, liked the previous approach more. If we cleaned everything and added the data back - the code would be cleaner, easier to maintain. 

But their main argument was that it could cause slowdowns on Meteor start.
Oswaldo was refactoring the server side for half a year to use this approach (among other things), it's not that we can easily turn it off now.

**Comment by antobinary** [comment]
From what I recall a major concern was that changing server side code would restart Meteor and would require logging in.
Either way, this is more of a discussion thread. No work will be done before revealing all sides

**Comment by oswaldoacauan** [comment]
> From what I recall a major concern was that changing server side code would restart Meteor and would require logging in.

Thats exactly why we not clear everything anymore, but we could change that and just reauthenticate the user on reconnect.

> Oswaldo was refactoring the server side for half a year to use this approach (among other things), it's not that we can easily turn it off now.

The refactor was mainly for code sanity and restructure, this behavior was only changed because of the user reconnect mentioned above. It should be pretty straightforward to clean all data on start, we just need to revisit the reconnect case. You can actually see that it works by doing `meteor reset` before the `npm start`.

> Perhaps I am seeing issues with clearing meetings that have ended. The symptom I see is high count of meetings and offline users on a server with few running meetings. I see the benefit on preserving state on a development environment (while making server side changes for example) but do not see it on production. Please comment.

This is probably related to how we are handling some Meeting event since its there where we remove a Meeting and its related data.

-----------

We already clear all data if in a production environment [`meteor reset`](https://github.com/bigbluebutton/bigbluebutton/blob/master/bigbluebutton-html5/package.json#L10) does exactly that so I miss the point where the changes we made on the refactor would be causing this issue.

**Comment by OZhurbenko** [comment]
@oswaldoacauan as far as I remember `meteor reset` doesn't remove the data from the real, not Meteor's built-in Mongo.
I played with this about half a year ago, and I remember I had to drop the db manually from the mongo terminal.
`meteor reset` only cleans the local Meteor's database, which of course would work in development and in the production simulator (which you are pointing to), but not when you deploy a bundle and use a standalone Mongo.
